Source/DafnyDriver/DafnyDriver.cs

```diff
@@ -145,14 +145,17 @@ public static CommandLineArgumentsResult ProcessCommandLineArguments(string[] ar
         Console.WriteLine("--------------------");
       }
 
+      ISet<String> filesSeen = new HashSet<string>();
       foreach (string file in CommandLineOptions.Clo.Files)
       { Contract.Assert(file != null);
         string extension = Path.GetExtension(file);
         if (extension != null) { extension = extension.ToLower(); }
 
         bool isDafnyFile = false;
         try {
-          dafnyFiles.Add(new DafnyFile(file));
+          var df = new DafnyFile(file);
+          if (!filesSeen.Add(df.CanonicalPath)) continue; // silently ignore duplicate
```

Comment:
Add curly braces around the `continue;`.

Reply:
Done